Dockerfile

```diff
@@ -22,6 +22,7 @@ RUN npm ci
 COPY . /app
 RUN npm run build
 RUN npm run test
+RUN chmod +x /app/entrypoint.sh
```

Comment:
Ouch I missed the entrypoint update, thank you for reviewing this. My idea was for it to look something like this:

```bash
#!/bin/sh

set -e
npm run migrate up &&
exec node --trace-warnings --abort-on-uncaught-exception --unhandled-rejections=strict dist/index.js
```

for that same reason. BUT let me check how calling node-pg-migrate from service.ts looks (cc: @LautaroPetaccio with the [signature server](https://github.com/decentraland/signatures-server/pull/5/files) )

Reply:
ouch x2, the default dockerfile were carefully tested. I recommend to roll that back x2 then

Reply:
It seems possible to be ready to be used programatically https://salsita.github.io/node-pg-migrate/#/api

Reply:
Ok so i rollbacked everything and added this, which works:

```ts
// service.ts:

  // migrate database
  const connectionString = await config.requireString(
    'PG_COMPONENT_PSQL_CONNECTION_STRING'
  )
  await runner({
    databaseUrl: connectionString,
    dir: path.resolve(__dirname, './migrations'),
    migrationsTable: 'pgmigrations',
    ignorePattern: '.*\\.map',
    direction: 'up',
  })

  // start components (server listener, db, etc...)
  await startComponents()
```

I have two suggestions:
- we move this to [pg-component](https://github.com/well-known-components/pg-component) maybe as an optional options object
```ts
// FROM:
  const pg = await createPgComponent({ logs, config, metrics }/*, { pool options here }*/)

// TO: 
  const pg = await createPgComponent({ logs, config, metrics }, {
  poolOptions: {},
  migrateOptions: {
    databaseUrl: connectionString,
    dir: path.resolve(__dirname, './migrations'),
    migrationsTable: 'pgmigrations',
    ignorePattern: '.*\\.map',
    direction: 'up',
  }
})
```
(we can drop "options" from the prop names). The bad thing here is forcing the dependency to the component user even if she's not running migrations. Maybe we can use a peer-dep

- I create a `migrations` port here which runs the migrations on `start`. The bad thin here is having to repeat this process on each client that needs it

what do you think? any other option I'm missing?

Reply:
Hi @menduz, the bash command does not forward the signals, but doing an `exec` on the node server does, as it replaces the process with the server one. The only issue that I see is that if a signal is sent while the migration is being done, the migration process would not receive the signal.

The thing about using a bash script is that we're splitting the running of the migrations from the server, as it is independent and, to, for example, develop locally, a flag must be set to prevent the migrations to run every time you run the server.

Reply:
Sorry I answer so late. I read your comment @LautaroPetaccio and I agree, **BUT** the lib will not run migrations each time. It checks against the database if there's something to run and continues on without doing anything (other than a query).

In my totally anecdotical and little practical tests it takes ~45ms to do so on each startup

Reply:
The thing is that the lib will connect to the DB and run a query to check which migrations were run each time the server is started. This is kind of a mixed behavior as IMO the server should not responsible of the migrations.
Saying so, if you are agains having a bash entrypoint, we should create a well known component for migrations so we use it everywhere.